core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala

```diff
@@ -1143,17 +1154,120 @@ class GroupCoordinatorTest {
       initialRebalanceResult.generation + 1,
       Set(leaderInstanceId, followerInstanceId, newMemberInstanceId),
       groupId,
-      CompletingRebalance)
+      CompletingRebalance,
+      Some(protocolType))
 
     checkJoinGroupResult(newFollowerResult,
       Errors.NONE,
       initialRebalanceResult.generation + 1,
       Set.empty,
       groupId,
       CompletingRebalance,
+      Some(protocolType),
       expectedLeaderId = newLeaderResult.memberId)
   }
 
+  @Test
+  def testProtocolTypeInJoinAndSyncRequest(): Unit = {
+    // JoinGroup(leader): The Protocol Type is None when there is an error
+    EasyMock.reset(replicaManager)
+    var leaderResponseFuture = sendJoinGroup(groupId, "fake-id", protocolType,
+      protocolSuperset, leaderInstanceId, DefaultSessionTimeout)
+
+    var leaderJoinGroupResult = await(leaderResponseFuture, 1)
+    assertEquals(Errors.UNKNOWN_MEMBER_ID, leaderJoinGroupResult.error)
+    assertEquals(None, leaderJoinGroupResult.protocolType)
+
+    // JoinGroup(leader): The Protocol Type is Defined when there is not error
+    EasyMock.reset(replicaManager)
+    leaderResponseFuture = sendJoinGroup(groupId, JoinGroupRequest.UNKNOWN_MEMBER_ID, protocolType,
+      protocolSuperset, leaderInstanceId, DefaultSessionTimeout)
+
+    // JoinGroup(follower): The Protocol Type is Defined when there is not error
```

Comment:
nit: I'd suggest making each one of these "stanzas" a separate test case

Reply:
You're right. Let me separate them.